de.metas.contracts/src/main/java/de/metas/contracts/inoutcandidate/SubscriptionShipmentScheduleHandler.java

```diff
@@ -99,14 +100,31 @@
 		documentLocationBL.setBPartnerAddress(documentLocation);
 
 		final I_C_Order order = create(term.getC_Order_Term(), I_C_Order.class);
-		newSched.setDeliveryRule(order.getDeliveryRule());
-		newSched.setDeliveryViaRule(order.getDeliveryViaRule());
+		if (order == null)
+		{
+			newSched.setDeliveryRule(term.getDeliveryRule());
+			newSched.setDeliveryViaRule(term.getDeliveryViaRule());
+		}
+		else
+		{
+			newSched.setDeliveryRule(order.getDeliveryRule());
+			newSched.setDeliveryViaRule(order.getDeliveryViaRule());         
+		}
+		
 
 		newSched.setQtyOrdered(subscriptionLine.getQty());
 		newSched.setQtyOrdered_Calculated(subscriptionLine.getQty());
 		newSched.setQtyReserved(subscriptionLine.getQty());
 
-		newSched.setLineNetAmt(newSched.getQtyReserved().multiply(term.getC_OrderLine_Term().getPriceActual()));
+		final  I_C_OrderLine orerLine =  term.getC_OrderLine_Term();
+		if (orerLine == null)
```

Comment:
same point I made about `C_Order_Term` note that this code dates back very far, so even if it's modernized here and there, it might contain all sorts of crap